standards/java_coding_standards.md

```diff
@@ -263,3 +263,9 @@ Other javadocs tags can be used as required.
 - Both multi-line (`\* */`) and single line (`//`) comments may be used
 
 - Comments within the method body should, wherever possible, be restricted to single line comments
+
+## Standard applied
+
+This standard was first agreed on 10 Feb 2020.
+
+This standard was last updated on 10 Feb 2020.
```

Comment:
Yes I'd agree with having something pretty much exactly like this.

I was wondering about:

- having a heading of **History** or **Status** rather than **Standard applied**
- using the word **adopted** or **approved** rather than **agreed**
- for each change, having a quick not on _what_ had been changed

So, something a bit like:

> ## Status
>
> 10/02/2020 - This standard was formally adopted on 20 Feb 2020.
>
> 17/02/2020 - Updated to include adoption date.
>
> 24/02/2020 - Updated to cover other things we hadn't thought of.
>
> 24/02/2021 - Completely changed to use Standard X from our new friends at Oracle.

Or something along those lines.

Reply:
1) Think *Status* as the heading is a good shout
2) Think *adopted* is a much better term than agreed
3) TBH I'm pretty against the idea of a summarised history. Whilst these documents are version-controlled it's a single click to open the history and see both the commit message and the actual changes all with the actual date they were made, and by whom. If you want that detail, its not much effort to drill in and find it. Whereas I think it adds reasonable effort plus more scope for forgetting to add the summary, or getting the info incorrect.

If anyone else sees this, please weigh in with a 👍 or 👎 on what you think.

Reply:
I agree with you.

I didn't word it well and my example didn't really convey what I wanted either!

What I'd like to have is some way of including a sort of "major" history in the standard, as and when people felt it was helpful as I think it could be a good idea to signpost people to go and look at the history.

But it definitely isn't required yet so we probably don't need to worry about it for now? (YAGNI)

I'm happy with:

> ## Status
>
> This standard was formally adopted on 20 Feb 2021.
>

And if we want something in the future then we can do this:

> ## Status history
>
> 20/02/2020 - We defined our own internal standard based on industry best practice and our hands-on experience.
> 20/02/2021 - We decided to ditch that and just use the new standard that Oracle published instead.

But we don't necessarily need to discuss and agree that now.